onmt/transforms/tokenize.py

```diff
@@ -48,6 +53,24 @@ def add_options(cls, parser):
                        "sampling, and dropout probability for BPE-dropout. "
                        "(target side)")
 
+        # subword vocabulary restriction options:
+        group.add('-src_subword_vocab', '--src_subword_vocab',
```

Comment:
Not particularly related to the vocabulary stuff per se, but in the same scope: wouldn't it be a bit more user-friendly to have the ability to set the option only once if we want to share src/tgt and convert to `src/tgt_XX` when parsing?

Reply:
We can set only once for the case of shared: set all `src_*` and `shared_vocab` except when we use Vocabulary Restriction.
If we use Vocabulary Restriction, we should set both sides even if shared and all option is equal. Because, there may be use case where we want to share all, but restrict only src not tgt or vice vesa.